test/auth.spec.js

```diff
@@ -0,0 +1,338 @@
+var _ = require('lodash');
+
+function uiFactory(expectedCalls) {
+    var i = 0;
+    return {
+        input: function (message, defaultValue) {
+            if (i + 1 > expectedCalls.length) {
+                expect.fail('Unexpected call to ui.input: "' + message + '", defaultValue: ' + JSON.stringify(defaultValue))
+            }
+            var expectedCall = expectedCalls[i];
+            var toSatisfy =  {
+                method: expectedCall.method,
+                message: expectedCall.message
+            };
+            if (expectedCall.defaultValue) {
+                toSatisfy.defaultValue = expectedCall.defaultValue;
+            }
+            expect({
+                method: 'input',
+                message: message,
+                defaultValue: defaultValue
+            }, 'to satisfy', toSatisfy);
+            i += 1;
+            return expect.promise(function () {
+                return expectedCall.returnDefault ? defaultValue : expectedCall.returns;
+            })
+        },
+        confirm: function (message, defaultValue) {
+            if (i + 1 > expectedCalls.length) {
+                expect.fail('Unexpected call to ui.confirm: "' + message + '", defaultValue: ' + JSON.stringify(defaultValue))
+            }
+            var expectedCall = expectedCalls[i];
+            expect({
+                method: 'confirm',
+                message: message
+            }, 'to satisfy', {
+                method: expectedCall.method,
+                message: expectedCall.message
+            });
+            i += 1;
+            return expect.promise(function () {
+                return expectedCall.returnDefault ? defaultValue : expectedCall.returns;
+            })
+        },
+        log: function (level, message) {
+            var expectedCall = expectedCalls[i];
+            expect(expectedCall, 'to satisfy', {
+                method: 'log',
+                message: message
+            })
+            i += 1;
+            return;
+        }
+    }
+}
+
+describe('lib/auth', function () {
+    describe('encoded credentials', function () {
+        it('should encode a set of credentials', function () {
+            var encodedCredentials = auth.encodeCredentials({
+                username: 'foo',
+                password: 'bar'
+            });
+            return expect(encodedCredentials, 'to equal', 'Zm9vOmJhcg==');
+        });
+        it('should decode a set of credentials', function () {
+            var decodedCredentials = auth.decodeCredentials('Zm9vOmJhcg==');
+            return expect(decodedCredentials, 'to equal', {
+                username: 'foo',
+                password: 'bar'
+            })
+        });
+    })
+    describe('injectRequestOptions', function () {
+        var baseRequestOptions = {
+            url: 'https://registry.npmjs.org/'
+        };
+        var requestOptions;
+        beforeEach(function () {
+            requestOptions = _.extend({}, baseRequestOptions);
```

Comment:
I totally understand what you're saying about avoiding dependencies. It's just a reflex by now, for me, to reach after lodash when doing stuff like this. I'm completely fine with handrolling the methods in line, if we only use it a few times in a test suite. I pushed a commit removing the lodash dev dep.